### DISCUSSION TITLE
Static JS load issue- after Node version upgrade from 14.x.x to 16.x.x to resolve @ckeditor/ckeditor5-core@37.0.1 library node incompatible issue

### DISCUSSION THREAD

**Discussion by devendra-tarento** [Sunbird-inQuiry/Community#73] [Q&A]
We were using SunbirdEd release-5.1.1 from last couple of month and from last week we started facing below issue when doing client app package installation. 

```
13:56:50 warning protractor@7.0.0: We have news to share - Protractor is deprecated and will reach end-of-life by Summer 2023. To learn more and find out about other options please refer to this post on the Angular blog. Thank you for using and contributing to Protractor. https://goo.gle/state-of-e2e-in-angular
13:56:50 warning protractor > webdriver-manager > request@2.88.2: request has been deprecated, see https://github.com/request/request/issues/314213:56:50 warning tslint@6.1.3: TSLint has been deprecated in favor of ESLint. Please see https://github.com/palantir/tslint/issues/4534 for more information. 
13:56:51 [3/5] Fetching packages... 1
3:56:55 error @ckeditor/ckeditor5-core@37.0.1: The engine "node" is incompatible with this module. Expected version ">=16.0.0". Got "14.19.0" 
13:56:55 error Found incompatible module. 
13:56:55 info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command. 
13:56:55 Client and Server Build complete Took 127 seconds to complete. 
13:56:55 starting docker build 
13:57:05 Sending build context to Docker daemon 470.4MB

```

Looking at the error it seems that one of the external library **@ckeditor/ckeditor5-core** used in our editor got recent  update with **minimum node version 16.x.x**.

To fix this we upgraded the **node version to 16.x.x** and it worked well in local system but when we upgraded the **node version to 16.x.x** in our **Dev server**. We started facing **504 gateway timeout** issue for some of the static files (js, svg, etc) randomly.

**Is anyone else facing the same issue ? 
Is upgrading to node 16.x.x risk as of now ?
What could be the possible solution to this ?**

**Note:** I also checked latest code base of release-5.2.0 of SunbirdEd and found that **build.sh, Dockerfile and Dockerfile.Build** still have **node version 14.19.0** . Does it mean SunbirdEd still using node 14.x.x in their servers.

@pallakartheekreddy

**Comment by sajeshkayyath** [comment]
@devendra-tarento - thanks for reaching out. we are checking this issue further and have some findings as below.

@rajeevsathish - while debugging what we noticed is that the `@ckeditor/ckeditor5-core@37.0.1` is getting installed by this package `"@project-sunbird/ckeditor-build-font": "^1.0.8",` and not by the `"@project-sunbird/ckeditor-build-classic": "^4.1.3",`
On checking further, QuML editor is not dependent on the package `@project-sunbird/ckeditor-build-font` and may be used in the Sunbird ED portal directly. Can you please check this further?

cc @vinukumar-vs @kameshbhr @PoojaJManjooran @vaibhavbhuva @rajnishdargan @Krishnaj20
